text/0125-xcm-asset-metadata.md

```diff
@@ -0,0 +1,226 @@
+
+The example use case of this instruction is to ask the reserve location of the asset to modify the metadata. So that, the original asset's metadata is updated according to the reserve location's rules.
+
+```rust
+ModifyMetadata {
+    /// The ID of an asset (a collection, fungible or nonfungible).
+    asset_id: AssetId,
+
+    /// The ID of an asset instance.
+    ///
+    /// If the value is `Undefined`, the modification request targets the collection, not a token.
+    instance: AssetInstance,
+
+    /// The map contains the keys mapped to the requested new values.
+    modification: MetadataMap,
+}
+```
+
+### Repurposing `AssetInstance::Undefined`
+
+As the new instructions show, this RFC reframes the purpose of the `Undefined` variant of the `AssetInstance` enum.
+This RFC proposes to use the `Undefined` variant of a collection identified by an `AssetId` as a synonym of the collection itself. I.e., an asset `Asset { id: <AssetId>, fun: Undefined }` is considered an NFT representing the collection itself.
+
+As a singleton non-fungible instance is barely distinguishable from its collection, this convention shouldn't cause any problems. 
+
+Thus, the `AssetInstance` docs must be updated accordingly in the implementations.
+
+## Drawbacks
+
+Regarding ergonomics, no drawbacks were noticed.
+
+As for the user experience, it could discover new cross-chain use cases involving asset collections and NFTs, indicating a positive impact.
+
+There are no security concerns except for the `ReportMetadata` instruction, which implies that the source of the information must be trusted.
+
+In terms of performance and privacy, there will be no changes.
+
+## Testing, Security, and Privacy
+
+The implementations must honor the contract for the new instructions. Namely, if the `instance` field has the value of `AssetInstance::Undefined`, the metadata must relate to the asset collection but not to a non-fungible token inside it.
+
+## Performance, Ergonomics, and Compatibility
+
+### Performance
+
+No significant impact.
+
+### Ergonomics
+
+Introducing a standard metadata format and a way of communicating it is a valuable addition to the XCM format that potentially increases cross-chain interoperability without the need to form ad-hoc chain-to-chain integrations via `Transact`.
+
+### Compatibility
+
+This RFC proposes new functionality, so there are no compatibility issues.
+
+## Prior Art and References
+
+[RFC: XCM Asset Metadata](https://github.com/polkadot-fellows/xcm-format/pull/50)
+
+## Unresolved Questions
+
+Should the `MetadataMap` and `MetadataKeys` be bounded, or is it enough to rely on the fact that every XCM message is itself bounded?
```

Comment:
No need to be bounded by the spec, will be bounded in practice by implementers using encoding/decoding bounds.
Same way `Assets` specifies an unbounded `vec<Asset>`, but in practice there are [hardcoded limits enforced](https://github.com/paritytech/polkadot-sdk/blob/dd9924fad0cd9314f21eff621208dc2ec60e0cf5/polkadot/xcm/src/v4/asset.rs#L553).